### PR TITLE
chore: drop mention of Python 2.7 from templates

### LINKED ISSUE #1118: [Python] Remove Python 2.7 from templates
It's probably safe to remove `2.7` support for all libraries now?

https://github.com/googleapis/synthtool/blob/bb5ef5a99df6d439c63fddf0f7b6fe58e0679b10/synthtool/gcp/templates/python_samples/noxfile.py.j2#L91

### PR DESCRIPTION
**Pull request by tseaver**
Closes #1118.

### CODE REVIEW

File: synthtool/gcp/common.py [hunk 1]
```diff
@@ -225,17 +225,9 @@ def py_library(self, **kwargs) -> Path:
             kwargs["default_python_version"] = "3.8"
         if "unit_test_python_versions" not in kwargs:
             kwargs["unit_test_python_versions"] = ["3.6", "3.7", "3.8", "3.9"]
-            if "microgenerator" not in kwargs:
```
Comment (parthea) [inline synthtool/gcp/common.py:228]: After this change, `microgenerator` is only used [here](https://github.com/googleapis/synthtool/blob/5f330a45e76ad2b3551c5276aaa818de324632b5/synthtool/gcp/templates/python_library/noxfile.py.j2). Could we remove `microgenerator` altogether?

File: synthtool/gcp/templates/python_library/noxfile.py.j2 [hunk 2]
```diff
@@ -193,7 +193,7 @@ def system(session):
         )
 
 {% if samples_test %}
```
Comment (parthea) [inline synthtool/gcp/templates/python_library/noxfile.py.j2:195]: `samples_test` doesn't appear to be used. Can we remove everything under this code block?

File: synthtool/gcp/templates/python_samples/noxfile.py.j2 [hunk 3]
```diff
@@ -89,7 +89,7 @@ def get_pytest_env_vars() -> Dict[str, str]:
 
 # DO NOT EDIT - automatically generated.
 # All versions used to tested samples.
```
Comment (parthea) [inline synthtool/gcp/templates/python_samples/noxfile.py.j2:91]:
```suggestion
# All versions used to test samples.
```

**Review by parthea — APPROVED** [review]
Added minor observations, otherwise LGTM.

File: synthtool/gcp/templates/python_library/noxfile.py.j2 (hunk 2 above)
Comment (tseaver) [reply synthtool/gcp/templates/python_library/noxfile.py.j2:195]: Dunno.  All the handwritten clients have separate noxfile(s) for samples these days:  none of them have a `samples` session.  @tswast added that stanza a couple of years back in 8a4579f2276d4814ec2262fc2078f6226da50a63: maybe he can recall if / how it ever got used.

File: synthtool/gcp/common.py (hunk 1 above)
Comment (tseaver) [reply synthtool/gcp/common.py:228]: The following repos do *not* have the `asyncmock` / `pytest-asyncio` test dependencies installed in their `noxfile.py`:

```bash
$ for noxfile in python-*/noxfile.py; do grep -q asyncmock $noxfile || echo $noxfile; done
python-bigquery/noxfile.py
python-bigquery-sqlalchemy/noxfile.py
python-cloud-core/noxfile.py
python-crc32c/noxfile.py
python-irm/noxfile.py
python-ndb/noxfile.py
python-spanner-django/noxfile.py
python-storage/noxfile.py
python-test-utils/noxfile.py
```

Of those these repos:

- `python-bigquery`: `owlbot.py` excludes `noxfile.py` from copying
- `python-bigquery-sqlalchemy`: pins explicit `{unit,system}_test_python_versions`, has no `async` support
- `python-cloud-core`: `owlbot.py` excludes `noxfile.py` from copying
- `python-crc32c`: has no `owlbot.py` (or `synth.py`)
- `python-irm`: repo is abandoned
- `python-ndb`: uses `templated_files` only for `.kokoro/`
- `python-spanner-django`: uses `templated_files` only for `.kokoro/` / `.trampolinerc`
- `python-storage`: `owlbot.py` excludes `noxfile.py` from copying
- `python-test-utils`: `owlbot.py` excludes `noxfile.py` from copying

Comment (parthea) [reply synthtool/gcp/common.py:228]: @tseaver My initial thought is we should be ok removing `microgenerator`. If we encounter any issue, we can add repo specific customizations in owlbot.py. WDYT?

File: synthtool/gcp/templates/python_library/noxfile.py.j2 (hunk 2 above)
Comment (parthea) [reply synthtool/gcp/templates/python_library/noxfile.py.j2:195]: I'm not getting any hits on [github search](https://github.com/search?q=org%3Agoogleapis+samples_test&type=code) for `samples_test`. Let's proceed and add repo specific customizations in owlbot.py if we notice any impact in downstream repos.

File: synthtool/gcp/common.py (hunk 1 above)
Comment (tseaver) [reply synthtool/gcp/common.py:228]: 39e8fc7

File: synthtool/gcp/templates/python_library/noxfile.py.j2 (hunk 2 above)
Comment (tseaver) [reply synthtool/gcp/templates/python_library/noxfile.py.j2]: a59fb21

**Comment by tseaver** [comment]
Hmm, @parthea I'm surprised your review is not sufficient to allow a merge.  Because I touched `synthtool/gcp/common.py`, which falls outside of the Python templates, it appears that I need one of the following approvals:

- @googleapis/github-automation
- @SurferJeffAtGoogle 
- @bcoe 
- @chingor13 

I would've expected that you were a member of @googleapis/github-automation.

**Comment by parthea** [comment]
@bcoe Please can you take a look for @googleapis/github-automation?

**Review by bcoe — APPROVED** [review]

**Pull request merged**